clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -143,6 +145,88 @@ private void setupCoordinator(int retryBackoffMs, int rebalanceTimeoutMs, Option
                                                 mockTime);
     }
 
+    @Test
+    public void testMetrics() {
+        setupCoordinator();
+
+        metrics.sensor("heartbeat-latency").record(1.0d);
+        metrics.sensor("heartbeat-latency").record(2.0d);
+        metrics.sensor("heartbeat-latency").record(6.0d);
+
+        assertNotNull(getMetric("heartbeat-response-time-max"));
+        assertNotNull(getMetric("heartbeat-rate"));
+        assertNotNull(getMetric("heartbeat-total"));
```

Comment:
Should we `assertNull` for all three before anything is recorded to check that they are created lazily? (similar below)

Reply:
Also, does the below `assertEquals` not subsume the `assortNotNull` ? Seems redundant to check for not-null first.

Reply:
The metrics are actually created eagerly :) And ack on the ordering of the record calls.